test-case/check-keyword-detection-in-D0i3.sh

```diff
@@ -0,0 +1,164 @@
+OPT_PARM_lst['p']=1                  OPT_VALUE_lst['p']=2100
+
+OPT_OPT_lst['s']='history-depth'     OPT_DESC_lst['s']='draining size'
+OPT_PARM_lst['s']=1                  OPT_VALUE_lst['s']=2100
+
+OPT_OPT_lst['b']='buffer'            OPT_DESC_lst['b']='buffer size'
+OPT_PARM_lst['b']=1                  OPT_VALUE_lst['b']=67200
+
+OPT_OPT_lst['l']='loop'              OPT_DESC_lst['l']='loop count'
+OPT_PARM_lst['l']=1                  OPT_VALUE_lst['l']=1
+
+OPT_OPT_lst['d']='duration'          OPT_DESC_lst['d']='interrupt kwd pipeline in # seconds'
+OPT_PARM_lst['d']=1                  OPT_VALUE_lst['d']=10
+
+func_opt_parse_option "$@"
+
+tplg=${OPT_VALUE_lst['t']}
+loop_cnt=${OPT_VALUE_lst['l']}
+buffer_size=${OPT_VALUE_lst['b']}
+history_depth=${OPT_VALUE_lst['s']}
+preamble_time=${OPT_VALUE_lst['p']}
+duration=${OPT_VALUE_lst['d']}
+
+[[ ${OPT_VALUE_lst['s']} -eq 1 ]] && func_lib_start_log_collect
+
+func_pipeline_export "$tplg" "kpbm:any"
+func_lib_setup_kernel_last_line
+
+if test "$PIPELINE_COUNT" != "1"; then
+    die "detected $PIPELINE_COUNT wov pipeline(s) from topology, but 1 is needed"
+fi
+
+# parser the parameters of the wov pipeline
+channel=$(func_pipeline_parse_value 0 ch_max)
+rate=$(func_pipeline_parse_value 0 rate)
+dev=$(func_pipeline_parse_value 0 dev)
+fmts=$(func_pipeline_parse_value 0 fmts)
+
+test_dir="/tmp"
+def_blob="$test_dir"/detertor_default_config.blob
+
+# get wov kcontrol ID
+wov_kctl_id=$(amixer controls |grep "Detector" |awk -F= '{print $2}' |cut -d"," -f1)
+
+# get wov PGA volume kcontrol name
+wov_kctl_pga=$(amixer controls |grep "KWD Capture Volume" |awk -F= '{print $4}')
+
+# adjust wov PGA volume
+amixer cset name="$wov_kctl_pga" 40
+
+# store the default config blob
+_store_default_wov_config_bolb()
+{
+    if [ ! -f "$def_blob" ]; then
+        sof-ctl -Dhw:0 -n "$wov_kctl_id" -r -o "$def_blob"
+        dlogi "default blob:"
+        cat "$def_blob"
+    fi
+}
+
+# get default preamble time and history depth
+_get_default_pt_hd()
+{
+    def_pt=$(awk -F "," '{print $4}' < "$def_blob")
+    def_hd=$(awk -F "," '{print $7}' < "$def_blob")
+}
+# restore the default config blob
+_restore_default_wov_config_bolb()
+{
+    if [ -f "$def_blob" ]; then
+        sof-ctl -Dhw:0 -n "$wov_kctl_id" -r -s "$def_blob"
+    fi
+}
+
+# update the blob
+_update_blob(){
+    new_blob=$test_dir"/new_blob"
+    [ "$preamble_time" -ge "$history_depth" ] || {
+        die "Warning: invalid arguments, preamble_time must be greater than or equal to history_depth"
+    }
+    awk -F, -v OFS=, '{if ( $4 == '"$def_pt"' && $7 == '"$def_hd"' ) $4='"$preamble_time"'; $7='"$history_depth"'}1' \
+        "$def_blob" > "$new_blob"
+    dlogi "kwd config blob is updated to:"
+    cat "$new_blob"
+    dlogi "write back the new kwd config blob"
+    sof-ctl -Dhw:0 -n "$wov_kctl_id" -r -s "$new_blob" >> /dev/null || die "Failed to write back the new kwd config bolb"
```

Comment:
Nit: `> /dev/null`  is enough and the common practice.

You can split such long lines very easily, the shall auto escapes new lines after `||`
```shell
sof-ctl -Dhw:0 -n "$wov_kctl_id" -r -s "$new_blob" >> /dev/null ||
     die "Failed to write back the new kwd config bolb"
```
How about packing some more information in the error message?
```shell
die "Failed to write back $new_blob ..."
```